src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java

```diff
@@ -69,10 +73,42 @@ public User createUser(User newUser) {
     return newUser;
   }
 
+    /**
+     * Returns max{userAge-3, 18}, used for initializing accounts
+     * @param userBirthday: birthday of the zser
+     * @return int max{userAge-3, 18}
+     */
+  private int findMinAgeDefault(Date userBirthday){
+      Date currentDate = new Date();
+      Calendar calendarUserBirthday = Calendar.getInstance();
+      calendarUserBirthday.setTime(userBirthday);
+      Calendar calendarCurrentDate = Calendar.getInstance();
+      calendarUserBirthday.setTime(currentDate);
+      int userAge = calendarCurrentDate.get(Calendar.YEAR) - calendarUserBirthday.get(Calendar.YEAR);
+      if(userAge - 3 < 18){
+          return 18;}
+      return userAge -3;
+  }
+
+    /**
+     * Used for initializing account
+     * @param userBirthday
+     * @return userage + 3
+     */
+  private int findMaxAgeDefault(Date userBirthday){
+      Date currentDate = new Date();
+      Calendar calendarUserBirthday = Calendar.getInstance();
+      calendarUserBirthday.setTime(userBirthday);
+      Calendar calendarCurrentDate = Calendar.getInstance();
+      calendarUserBirthday.setTime(currentDate);
+      int userAge = calendarCurrentDate.get(Calendar.YEAR) - calendarUserBirthday.get(Calendar.YEAR);
+      return userAge +3;
+  }
+
```

Comment:
First, if I'm born in September 2000 then you would falsely assume that I'm 22, even though I'm still 21 and give me the default minAge of 18 instead of 19. Secondly, you write alot of code twice. 
I propose changing it to the following:
```
  private long getAgeInMilliSeconds(Date birthday) {
    return new Date().getTime() - birthday.getTime();
  }
  
  private int getAge(Date birthday) {
    Calendar c = Calendar.getInstance();
    c.setTime(new Date());
    int currentYear = c.get(Calendar.YEAR);
    c.setTime(birthday);
    int birthYear = c.get(Calendar.YEAR);
    return currentYear - birthYear;
  }

  private int findMinAgeDefault(Date userBirthday){
    long age = getAgeInMilliSeconds(userBirthday);
    return age - 3 * Time.ONE_YEAR < 18 * Time.ONE_YEAR ? 18 : getAge(userBirthday) - 3;
  }
  
  private int findMaxAgeDefault(Date userBirthday) {
    return getAge(userBirthday) + 3;
  }
```
You can add  `  public static final long ONE_YEAR = 31536000000L;
` in the Time File for this.
But props for not using the deprecated date.getYear()!

Reply:
implemented you methods. Thanks